homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,310 @@
+                dist = location.distance(latitude, longitude,
+                                         network_latitude, network_longitude)
+                if dist < minimum_dist:
+                    minimum_dist = dist
+                    result = network[ATTR_ID]
+
+            return result
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Could not connect to CityBikes API endpoint")
+            return None
+        except ValueError:
+            _LOGGER.error("Received non-JSON data from CityBikes"
+                          " API endpoint")
+            return None
+        except vol.Invalid as err:
+            _LOGGER.error("Received unexpected JSON from CityBikes"
+                          " API endpoint: %s", err)
+            return None
+
+    def __init__(self, hass, network_id):
+        """Initialize the network object."""
+        self._hass = hass
+        self._network_id = network_id
+        self._stations = []
+        self._ready = False
+
+    @property
+    def uid(self):
+        """Return the network ID."""
+        return self._network_id
+
+    @property
+    def stations(self):
+        """Return the stations of the network."""
+        return self._stations
+
+    @property
+    def ready(self):
+        """Return the readiness state of the network object."""
+        return self._ready
+
+    @asyncio.coroutine
+    def async_refresh(self, now=None):
+        """Refresh the state of the network."""
+        try:
+            session = async_get_clientsession(self._hass)
+
+            with async_timeout.timeout(5, loop=self._hass.loop):
+                req = yield from session.get(DEFAULT_ENDPOINT.format(
+                    uri=STATIONS_URI.format(uid=self._network_id)))
+
+            json_response = yield from req.json()
+            network = STATIONS_RESPONSE_SCHEMA(json_response)
+            self._stations = network[ATTR_NETWORK][ATTR_STATIONS_LIST]
+            self._ready = True
+            return
+
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Could not connect to CityBikes API endpoint")
+        except ValueError:
+            _LOGGER.error("Received non-JSON data from CityBikes API"
+                          " endpoint")
+        except vol.Invalid as err:
+            _LOGGER.error("Received unexpected JSON from CityBikes API"
+                          " endpoint: %s", err)
+
+        self._ready = False
+
+
+class CityBikesStation(Entity):
+    """CityBikes API Sensor."""
+
+    def __init__(self, network, station_id, base_name=''):
+        """Initialize the sensor."""
+        self._network = network
+        self._station_id = station_id
+        self._station_data = {}
+        self._base_name = base_name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        try:
+            return self._station_data[ATTR_FREE_BIKES]
+        except KeyError:
```

Comment:
It can happen in the beginning, when `_station_data` is `None`. Anyway, I changed it to use `get` with a default specifier.